x/staging/subspaces/keeper/keeper.go

```diff
@@ -197,3 +197,22 @@ func (k Keeper) UnbanUserInSubspace(ctx sdk.Context, subspaceID, user, admin str
 	store.Delete(types.SubspaceBannedUserKey(subspaceID, user))
 	return nil
 }
+
+// CheckSubspaceUserPermission checks the permission of the given user inside the subspace with the
+// given id to make sure they are able to perform operations inside it
+func (k Keeper) CheckSubspaceUserPermission(ctx sdk.Context, subspaceID string, user string) error {
+	subspace, found := k.GetSubspace(ctx, subspaceID)
+	if !found {
+		return sdkerrors.Wrapf(types.ErrInvalidSubspaceID, "the subspace %s doesn't exist", subspaceID)
+	}
+
+	if k.IsBanned(ctx, subspaceID, user) {
+		return sdkerrors.Wrapf(sdkerrors.ErrInvalidRequest, "%s is banned from subspace %s", user, subspaceID)
+	}
+
+	if subspace.Type == types.SubspaceTypeClosed && !k.IsRegistered(ctx, subspaceID, user) {
+		return sdkerrors.Wrapf(sdkerrors.ErrInvalidRequest, "%s is not registered in subspace %s", user, subspaceID)
+	}
```

Comment:
I think we should define a new error inside the subspace to handle this cases. The error might be called something like `ErrPermissionDenied` so that it can be applied to both unregistered and banned users.

Then we will be able to change these lines with 

```suggestion
	if k.IsBanned(ctx, subspaceID, user) {
		return sdkerrors.Wrapf(types.ErrPermissionDenied, "%s is banned", user)
	}

	if subspace.Type == types.SubspaceTypeClosed && !k.IsRegistered(ctx, subspaceID, user) {
		return sdkerrors.Wrapf(types.ErrPermissionDenied, "%s is not registered", user)
	}
```

(No need to specify the subspace id each time)